docs/content/docs/build/containers.md

```diff
@@ -149,8 +149,9 @@ You should always check the `isDirty` flag before closing the container or navig
 
 A container is considered dirty in the following cases:
 
-1. The container has local changes that have not yet been acknowledged by the service. These unacknowledged changes will be lost if the container is closed.
-1. There is no network connection while making changes to the container. These changes cannot be acknowledged by the service until the network connection is restored.
+1. The container has been created in the detached state, and it has not been attached yet.
+2. The container was attached, but it has local changes that have not yet been acknowledged by the service. These unacknowledged changes will be lost if the container is closed.
+3. There is no network connection while making changes to the container. These changes cannot be acknowledged by the service until the network connection is restored.
```

Comment:
Isn't 3 a subset of 2? I'm not sure if we want to say that (maybe we do).
I think one think you can add - that state will flicker quickly between dirty and saved as changes are made and that host may choose to use some delay before showing that in UX - both as continuous status update, as well as raising "are you sure you want to close this doc? You have unsaved changes" kind of dialog on close.

Reply:
Agree, its a subset. This was an older language that differentiated the two - will make the change.

Interesting point about flickering. Are there UX use cases where we expose "saved"/"dirty" state continuously to user as they modify shared data? Id imagine we would still have quite a bit of flickering even in normal ops flow. Or these are mostly one-offs, for example on document close like you mentioned ?